extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/data/input/kafkainput/KafkaInputReader.java

```diff
@@ -113,14 +169,9 @@ private CloseableIterator<InputRow> buildBlendedRows(InputEntityReader valuePars
                 "Unsupported input format in valueFormat. KafkaInputFormat only supports input format that return MapBasedInputRow rows"
             );
           }
-          Map<String, Object> event = new HashMap<>(headerKeyList);
-          /* Currently we prefer payload attributes if there is a collision in names.
-              We can change this beahvior in later changes with a config knob. This default
-              behavior lets easy porting of existing inputFormats to the new one without any changes.
-            */
-          event.putAll(valueRow.getEvent());
-
-          HashSet<String> newDimensions = new HashSet<String>(valueRow.getDimensions());
+
+          final Map<String, Object> event = buildBlendedEventMap(valueRow.getEvent(), headerKeyList);
```

Comment:
Yeah, it is left like that intentionally for now because auto discovered dimensions are created with a string dimension indexer pointing at them today which is just going to end up calling java .toString on these values.

I am actually working on making it possible to use the nested column indexer for all auto discovered dimensions instead of the string dimension indexer we use today, which will be the appropriate time to add these columns to the list of discovered columns.

Reply:
Thanks for the clarification - that makes sense to me.